### PR TITLE
Use new sort code in window functions

### LINKED ISSUE #1608: Improve sorting for window functions
Follow-up to #1367 without the passive aggressive title @lnkuiper @hawkfish

### PR DESCRIPTION
**Pull request by lnkuiper**
This PR replaces the code under `SortCollectionForPartition` in `PhysicalWindow` with the new sorting code. It sinks the `DataChunk`s in a `ChunkCollection` into a `LocalSortState`, which then does a Radix Sort. Afterwards, the sorted `ChunkCollection` is recovered.

Eventually, we want to run the window functions over the `RowLayout`, but that's a bigger project. I figured this would give a good speedup for now.

### CODE REVIEW

**Comment by hannesmuehleisen** [comment]
Did you run some benchmarks?

**Review by Mytherin — APPROVED** [review]
Looks great! Also curious to see numbers :)

**Comment by lnkuiper** [comment]
I ran all the queries in TPC-DS that have a window function that depends on ordering on jewels07, 16 threads. Reported time is median time of 5 queries, and with `.mode trash` because we don't care about the output.

SF1
| Query | Old       | New       |
|-------|-----------|-----------|
| 36    | __0.050__ | __0.050__ |
| 44    | __0.035__ | __0.035__ |
| 47    | __0.523__ | 0.555     |
| 49    | __0.016__ | 0.017     |
| 51    | 3.217     | __3.077__ |
| 57    | __0.289__ | 0.334     |
| 67    | 1.856     | __1.165__ |
| 70    | 0.058     | __0.057__ |

SF10
| Query | Old        | New        |
|-------|------------|------------|
| 36    | __0.191__  | 0.196      |
| 44    | __0.127__  | 0.130      |
| 47    | 1.823      | __1.814__  |
| 49    | __0.016__  | __0.016__  |
| 51    | 32.743     | __30.426__ |
| 57    | 1.184      | __1.128__  |
| 67    | 37.317     | __15.975__ |
| 70    | __0.339__  | __0.339__  |

Not every query is faster, but where it's slower it's comparable, and where it's faster it can make a large difference (e.g. 67). Looking at the CPU usage during 67, 1 thread is busy for a long time at the end. This can of course be sped up if we enable multi-threaded sorting here.

File: src/include/duckdb/common/types/data_chunk.hpp [hunk 1]
```diff
@@ -86,6 +86,9 @@ class DataChunk {
 	DUCKDB_API void Copy(DataChunk &other, const SelectionVector &sel, const idx_t source_count,
 	                     const idx_t offset = 0) const;
 
+	//! Splits the DataChunk in two
```
Comment (hawkfish) [inline src/include/duckdb/common/types/data_chunk.hpp:89]: "Splits the DataChunk in two vertically (see Fuse)"?

**Review by hawkfish — APPROVED** [review]

**Pull request merged**